repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingFilter.java

```diff
@@ -56,22 +80,127 @@ public RateLimitingFilter(
-        configurationService.subscribeTo(filterConfig.getFilterName(), config, xsdURL, handlerFactory, RateLimitingConfiguration.class);
+    public void destroy() {
+        LOG.trace("Rate Limiting filter destroying...");
+        configurationService.unsubscribeFrom(configFilename, this);
+        LOG.trace("Rate Limiting filter destroyed.");
+    }
+
+    @Override
+    public void configurationUpdated(RateLimitingConfiguration configurationObject) throws UpdateFailedException {
+        RateLimitingConfig newConfig = new RateLimitingConfig();
+        newConfig.rateLimitingService = RateLimitingServiceFactory.createRateLimitingService(
+                new ManagedRateLimitCache(getDatastore(configurationObject.getDatastore())), configurationObject);
+        newConfig.describeLimitsUriRegex = configurationObject.getRequestEndpoint() != null ?
+                Optional.of(Pattern.compile(configurationObject.getRequestEndpoint().getUriRegex())) :
+                Optional.<Pattern>absent();
+        newConfig.includeAbsoluteLimits = configurationObject.getRequestEndpoint() != null &&
+                configurationObject.getRequestEndpoint().isIncludeAbsoluteLimits();
+        newConfig.isOverLimit429ResponseCode = configurationObject.isOverLimit429ResponseCode();
+        newConfig.datastoreWarnLimit = configurationObject.getDatastoreWarnLimit().intValue();
+
+        config.set(newConfig);
+        initialized = true;
+    }
+
+    /**
+     * For now, carry over the old behavior of building a new handler per request.
+     */
+    private RateLimitingHandler buildHandler() {
+        // snapshot of current config
+        RateLimitingConfig currentConfig = config.get();
+
+        return new RateLimitingHandler(
+                new RateLimitingServiceHelper(currentConfig.rateLimitingService, new ActiveLimitsWriter(), new CombinedLimitsWriter()),
+                eventService,
+                currentConfig.includeAbsoluteLimits,
+                currentConfig.describeLimitsUriRegex,
+                currentConfig.isOverLimit429ResponseCode,
+                currentConfig.datastoreWarnLimit);
+    }
+
+    private Datastore getDatastore(DatastoreType datastoreType) {
+        String requestedDatastore = datastoreType.value();
+        if (StringUtilities.isNotBlank(requestedDatastore)) {
+            LOG.info("Requesting datastore {}", datastoreType);
+
+            if (requestedDatastore.equals(DEFAULT_DATASTORE_NAME)) {
+                LOG.info("Using requested datastore {}", requestedDatastore);
+                return datastoreService.getDefaultDatastore();
+            }
+
+            Datastore datastore = datastoreService.getDatastore(requestedDatastore);
+            if (datastore != null) {
+                LOG.info("Using requested datastore {}", requestedDatastore);
+                return datastore;
+            } else {
+                LOG.warn("Requested datastore not found");
```

Comment:
Yeah, this method is copy/pasta as I didn't want to mess with it.

https://github.com/rackerlabs/repose/blob/master/repose-aggregator/components/filters/rate-limiting/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerFactory.java#L70